[shirsho-12] iP

shirsho-12 opened the pull request.

Commit:
Add Gradle support

shirsho-12 committed:
Level-1

shirsho-12 committed:
Level-2, indent fix

shirsho-12 committed:
Level 3 main task

shirsho-12 committed:
trailing input bug fix

shirsho-12 committed:
add classes

shirsho-12 committed:
refactor to multiple classes

shirsho-12 committed:
remove UI lines

shirsho-12 committed:
Task-5 complete

shirsho-12 committed:
Level-6

shirsho-12 committed:
enums

shirsho-12 committed:
UI fixes

shirsho-12 committed:
testing

shirsho-12 committed:
Complete save/load functionality

shirsho-12 committed:
Date and time formatting

shirsho-12 committed:
Merge branch 'branch-Level-7'

shirsho-12 committed:
Merge branch 'branch-Level-8'

shirsho-12 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

shirsho-12 committed:
Datetime file save fix

shirsho-12 committed:
Gradle fix

shirsho-12 committed:
Deadline bug fix

shirsho-12 committed:
Gradle checkstyle

shirsho-12 committed:
OOP increase

shirsho-12 committed:
Packaging - code refactor

shirsho-12 committed:
Checkstyle reformatting

shirsho-12 committed:
Task tests

shirsho-12 committed:
Command, Parser, TaskList and Ui tests

shirsho-12 committed:
JAR build support

shirsho-12 committed:
JavaDoc addition

shirsho-12 committed:
Coding standard fixes

shirsho-12 committed:
Find task command implementation

shirsho-12 committed:
Merge branch 'branch-Level-9'

shirsho-12 committed:
Merge branch 'branch-A-CodingStandard'

shirsho-12 committed:
Find case readded to parser

shirsho-12 committed:
Merge branch 'branch-A-JavaDoc'

shirsho-12 committed:
Additional Fixes

shirsho-12 committed:
GFMD

shirsho-12 committed:
More GFMD

Nafeij commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,32 @@
+ * The command to be executed. All commands are subclasses of this class.
+ */
+public abstract class Command {
+    protected TaskList taskList;
```
Perhaps these class variables could be declared `private final`, to better abide by the guidelines of Java information hiding and encapsulation.

shirsho-12 replied:
I made this protected so that it's available within the package, but I agree with your comment. I'll make it private.

Nafeij commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,32 @@
+        this.ui = ui;
+    }
+
+    public abstract boolean equals(Object obj);
```
Could you describe what these methods are supposed to do?

shirsho-12 replied:
As in what does Command() do? Or assign? I've added JavaDocs for assign. equals() is just the standard equals.

Nafeij commented on src/main/java/duke/parser/Action.java:
```diff
@@ -0,0 +1,25 @@
+        return action;
+    }
+
+    public static Action getAction(String action) {
```
Could you describe what this method is supposed to do?

shirsho-12 replied:
Umm it's just a getter, to my knowledge getters and setters don't need explicit JavaDocs.

Nafeij commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,107 @@
+ */
+public class Storage {
+
+    private String filePath = "data/duke.txt";
```
Can this be made a constant "`private static final`"?

shirsho-12 replied:
Agreed

Nafeij commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    public String getDeadline() {
+        return deadline.format(DateTimeFormatter.ofPattern("d/MM/yyyy HHmm"));
```
Perhaps `"d/MM/yyyy HHmm"` can be extracted and made a constant.

Nafeij commented on src/main/java/duke/parser/Action.java:
```diff
@@ -0,0 +1,25 @@
+package duke.parser;
+
+enum Action {
```
I like the way you used enums. Very thorough!

Nafeij reviewed (commented):
Your code could use more documentation in the form of header comments. Otherwise very neat and well-structured!

tingyic commented on src/main/java/duke/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,34 @@
+/**
+ * The command to add a deadline to the task list.
+ */
+public class AddDeadlineCommand extends Command {
```
Great use of OOP here! Applies for the other command functions. :)

redHat-arko commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,51 @@
+
+
+
+public class ParserTest {
```
Really appreciate the extensive testing you applied to your code by going above and beyond the minimal requirements

redHat-arko commented on src/main/java/duke/command/AddDeadlineCommand.java:
```diff
@@ -1,3 +1,8 @@
+package duke.command;
```
Really great use of packaging by going beyond the single-package requirement and packaging different components of the final product instead.

arkobrat commented on src/main/java/duke/command/AddEventCommand.java:
```diff
@@ -0,0 +1,33 @@
+package duke.command;
```
Great use of packaging here by going beyond the single-package requirement

arkobrat commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,51 @@
+import duke.task.ToDo;
+
+
+
```
Appreciate the extensive use of unit testing by going above and beyond the minimal requirements.

shirsho-12 committed:
JavaFx Ui step 1

shirsho-12 committed:
JavaFx Ui step 2

shirsho-12 committed:
JavaFx Ui step 3

shirsho-12 committed:
Bye command added, functional JavaFx

shirsho-12 committed:
Further checkstyle fixes

All checkstyle violations from adding JavaFx are fixed.
Previously, there were multiple violations due to the lack of
JavaDocs and code quality issues.

Let's fix these violations and ensure that the codebase follows
the checkstyle.

shirsho-12 committed:
Add assertions

A lack of assertions in the code causes bugs to creep through during
development. These bugs can make refactoring and adding new features
difficult.

Adding assertions prevents such problems from occuring in the first
place.

Let's add assertions in various sections of the codebase to ensure code
correctness.

shirsho-12 committed:
Fix code quality issues

shirsho-12 committed:
Merge pull request #1 from shirsho-12/branch-Assertions

Branch assertions

shirsho-12 committed:
Merge branch 'master' into A-CodeQuality

shirsho-12 committed:
Merge pull request #2 from shirsho-12/A-CodeQuality

Fix code quality issues

shirsho-12 committed:
Set up GitHub Actions CI

shirsho-12 committed:
Add chronological sort

shirsho-12 committed:
Merge branch 'branch-C-Sort'

shirsho-12 committed:
Add more GFMD

shirsho-12 committed:
Update gradle.yml

Fix Java build bugs

shirsho-12 committed:
Fix ui tests

shirsho-12 committed:
Merge branch 'master' of https://github.com/shirsho-12/ip

shirsho-12 committed:
Add repo token to first-interaction

shirsho-12 committed:
Remove first interaction

First interaction only works on Ubuntu

shirsho-12 committed:
Update README.md

shirsho-12 committed:
Change images in UI

shirsho-12 committed:
Modify NoDuKo answers

shirsho-12 committed:
Add complete test coverage

shirsho-12 committed:
Merge branch 'master' of https://github.com/shirsho-12/ip

shirsho-12 committed:
Add user guide

shirsho-12 committed:
Fix test checkstyle issues

shirsho-12 committed:
Added finishing touches

shirsho-12 committed:
Class header checkstyle fixes

shirsho-12 committed:
File path fix
